src/main/java/org/terasology/metalrenegades/ai/system/AgentSpawnSystem.java

```diff
@@ -0,0 +1,113 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.metalrenegades.ai.system;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.entitySystem.entity.EntityBuilder;
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.prefab.Prefab;
+import org.terasology.entitySystem.prefab.PrefabManager;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterMode;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.entitySystem.systems.UpdateSubscriberSystem;
+import org.terasology.logic.location.LocationComponent;
+import org.terasology.metalrenegades.ai.component.AgentComponent;
+import org.terasology.metalrenegades.ai.component.HomeComponent;
+import org.terasology.metalrenegades.ai.component.PotentialHomeComponent;
+import org.terasology.registry.In;
+
+import java.util.Collection;
+
+/**
+ * Spawns new agents inside of available buildings with {@link PotentialHomeComponent}.
+ */
+@RegisterSystem(value = RegisterMode.AUTHORITY)
+public class AgentSpawnSystem extends BaseComponentSystem implements UpdateSubscriberSystem {
+
+    private static final Logger logger = LoggerFactory.getLogger(AgentSpawnSystem.class);
+
+    @In
+    private EntityManager entityManager;
+
+    @In
+    private PrefabManager prefabManager;
+
+    @Override
+    public void update(float delta) {
+        for (EntityRef entity : entityManager.getEntitiesWith(PotentialHomeComponent.class)) {
+            PotentialHomeComponent potentialHomeComponent = entity.getComponent(PotentialHomeComponent.class);
+            if(potentialHomeComponent.residents.size() >= potentialHomeComponent.maxResidents) {
+                continue;
+            }
+
+            EntityRef agent = spawnAgent(entity);
+            if (agent == null) { // if no entity was generated.
+                continue;
+            }
+
+            potentialHomeComponent.residents.add(agent);
+
+            entity.saveComponent(potentialHomeComponent);
+        }
+    }
+
+    /**
+     * Spawns a random agent inside the center of a provided building entity.
+     *
+     * @param homeEntity The building entity to spawn inside.
+     * @return The new agent entity, or null if spawning is not possible.
+     */
+    private EntityRef spawnAgent(EntityRef homeEntity) {
+        Prefab agentPrefab = chooseAgentPrefab();
+        if(agentPrefab == null) { // if no prefab is available.
```

Comment:
Run Checkstyle over your code, it'll highlight trivial code style issues like here - convention is to have a space between `if` and the starting `(` :-)

Reply:
I have a checkstyle plugin on intelliJ, but I guess a few of them slipped through, whoops! Fixed these where I missed them 👍